pkg/driver/controller_server.go

```diff
@@ -929,13 +923,15 @@ func (driver *Driver) controllerUnpublishVolume(volumeID string, nodeID string,
 		return err
 	}
 
+	// pass the nodeID to the container storage provider and do not do a look up of the node object as
+	// node object may have been deleted as part of UnloadNodeInfo when the node went down
 	if !existingVolume.Published {
-		log.Infof("Volume %s is already unpublished from node %s", existingVolume.Name, node.Name)
+		log.Infof("Volume %s is already unpublished from node with ID %s", existingVolume.Name, nodeID)
 		return nil
 	}
 
 	// Remove ACL from the volume based on the requested Node ID
-	err = storageProvider.UnpublishVolume(existingVolume.ID, node.UUID)
+	err = storageProvider.UnpublishVolume(existingVolume.ID, nodeID)
```

Comment:
Looks good.  Did you verify the CSP handles this idempotently?

Reply:
Yes. Also following are the sequence of events when rhel5 went down and pod moved to rhel6
```
   66874 Mar  2 2020 19:04:52 admin           Succeeded Remove ACL for volume pvc-e23752b1-1196-49bd-bb96-77aa8843fd0f                                      
   66875 Mar  2 2020 19:04:52 admin           Succeeded Delete initiator group container-node-hiqa-rhel5-fc                                                 
   66876 Mar  2 2020 19:04:52 admin           Succeeded Delete application server record container-node-hiqa-rhel5                                          
   66877 Mar  2 2020 19:04:53 admin           Succeeded Create application server record container-node-hiqa-rhel6                                          
   66878 Mar  2 2020 19:04:53 admin           Succeeded Create initiator group container-node-hiqa-rhel6-fc                                                 
   66879 Mar  2 2020 19:04:53 admin           Succeeded Add initiator(s)  to initiator group container-node-hiqa-rhel6-fc                                   
   66880 Mar  2 2020 19:04:53 admin           Succeeded Add initiator(s)  to initiator group container-node-hiqa-rhel6-fc                                   
   66881 Mar  2 2020 19:04:53 admin           Succeeded Add ACL for volume pvc-e23752b1-1196-49bd-bb96-77aa8843fd0f                                         
```